src/python/pants/engine/internals/graph.py

```diff
@@ -1205,8 +1207,24 @@ def __init__(
 
 
 @rule
-async def determine_explicitly_provided_dependencies(
+async def convert_dependencies_request_to_explicitly_provided_dependencies_request(
     request: DependenciesRequest,
+) -> ExplicitlyProvidedDependenciesRequest:
+    """This rule discards any deps predicate from DependenciesRequest.
+
+    Calculating ExplicitlyProvidedDependencies does not use any deps traversal predicates as it is
+    meant to list all explicit deps from the given field. By stripping the predicate from the
+    request, we ensure that the cache key for ExplicitlyProvidedDependencies calculation does not
+    include the predicate increasing the cache-hit rate.
+    """
+    # TODO: Maybe require Get(ExplicitlyProvidedDependencies, ExplicitlyProvidedDependenciesRequest)
+    #       and deprecate Get(ExplicitlyProvidedDependencies, DependenciesRequest) via this rule.
+    return ExplicitlyProvidedDependenciesRequest(request.field)
```

Comment:
In #19155, I just had a comment on the `determine_explicitly_provided_dependencies` rule that said: `NB: This rule ignores request.should_traverse_deps_predicate.`

@thejcannon recommended I use an actual type instead of just a comment in https://github.com/pantsbuild/pants/pull/19155#discussion_r1221803528
> We might consider a different input type that doesn't have the ignored attribute?

So, I added this `ExplicitlyProvidedDependenciesRequest` with a rule to minimize API churn in this PR.

Do we want to deprecate `Get(ExplicitlyProvidedDependencies, DependenciesRequest)`? If/when we do that, we can probably put a deprecation warning in this rule.

Reply:
I think that's a good idea, although feel free to carry this over into a future PR so we can merge this one.

Also up to you here :smile:

Reply:
I think we can deprecate once we have updated all uses within the pants repo. So, it fits best in a future PR.